pkg/apiserver/rest/dualwriter.go

```diff
@@ -117,3 +122,63 @@ func (u *updateWrapper) Preconditions() *metav1.Preconditions {
 func (u *updateWrapper) UpdatedObject(ctx context.Context, oldObj runtime.Object) (newObj runtime.Object, err error) {
 	return u.updated, nil
 }
+
+func SetDualWritingMode(
+	ctx context.Context,
+	kvs *kvstore.NamespacedKVStore,
+	features featuremgmt.FeatureToggles,
+	entity string,
+	legacy LegacyStorage,
+	storage Storage,
+) (DualWriter, error) {
+	toMode := map[string]DualWriterMode{
+		"1": Mode1,
+		"2": Mode2,
+		"3": Mode3,
+		"4": Mode4,
+	}
+	errDualWriterSetCurrentMode := errors.New("failed to set current dual writing mode")
+
+	// Use entity name as key
+	m, ok, err := kvs.Get(ctx, entity)
+	if err != nil {
+		return nil, errors.New("failed to fetch current dual writing mode")
+	}
+	if !ok {
+		// Default to mode 1
+		m = fmt.Sprint(Mode1)
+		err := kvs.Set(ctx, entity, m)
+		if err != nil {
+			return nil, errDualWriterSetCurrentMode
+		}
+	}
+
+	currentMode := toMode[m]
```

Comment:
we should probably make sure that toMode[m] exists and default to mode1 if it doesn't

Reply:
Good point. Added a check in [dc147b9](https://github.com/grafana/grafana/pull/87919/commits/dc147b994fea5a3423cf3568c6114b8d35e8bf25).